src/compiler.jl

```diff
@@ -609,25 +608,50 @@ function apply_latest_rev(B::LLVM.API.LLVMBuilderRef, OrigCI::LLVM.API.LLVMValue
     return nothing
 end
 
+function emit_error(B::LLVM.Builder, string)
+    curent_bb = position(B)
+    fn = LLVM.parent(curent_bb)
+    mod = LLVM.parent(fn)
+    ctx = context(mod)
+
+    # 1. get the error function
+    if haskey(functions(mod), "jl_error")
+        jlerror_func = functions(mod)["jl_error"]
+    else
+        jlerror_T = LLVM.FunctionType(LLVM.VoidType(ctx), LLVMType[LLVM.PointerType(LLVM.Int8Type(ctx))])
+        jlerror_func = LLVM.Function(mod, "jl_error", jlerror_T)
+        push!(function_attributes(jlerror_func), EnumAttribute("noreturn"; ctx))
+    end
+
+    # 2. Call error function and insert unreachable
+    call!(B, jlerror_func, LLVM.Value[globalstring_ptr!(B, string)])
+    unreachable!(B)
+
+    # 3. Change insertion point so that we don't stumble later
+    # FIXME(@wsmoses): It seems like Enzyme does not use the `B` passed here
```

Comment:
@wsmoses One can't create basic blocks since the builder afterwards doesn't know where to continue.